cmake/emu/qemu.cmake

```diff
@@ -226,15 +226,50 @@ endif(QEMU_PIPE_STACK)
 if(CONFIG_X86_64)
   # QEMU doesn't like 64-bit ELF files. Since we don't use any >4GB
   # addresses, converting it to 32-bit is safe enough for emulation.
-  set(QEMU_KERNEL_FILE "${CMAKE_BINARY_DIR}/zephyr-qemu.elf")
-  add_custom_target(qemu_kernel_target
+  add_custom_target(qemu_image_target
     COMMAND
     ${CMAKE_OBJCOPY}
     -O elf32-i386
     $<TARGET_FILE:${logical_target_for_zephyr_elf}>
     ${CMAKE_BINARY_DIR}/zephyr-qemu.elf
     DEPENDS ${logical_target_for_zephyr_elf}
     )
+
+  # Split the 'locore' and 'main' memory regions into separate executable
+  # images and specify the 'locore' as the boot kernel, in order to prevent
+  # the QEMU direct multiboot kernel loader from overwriting the BIOS and
+  # option ROM areas located in between the two memory regions.
+  # (for more details, refer to the issue zephyrproject-rtos/sdk-ng#168)
+  add_custom_target(qemu_locore_image_target
+    COMMAND
+    ${CMAKE_OBJCOPY}
+    -j .locore
+    ${CMAKE_BINARY_DIR}/zephyr-qemu.elf
+    ${CMAKE_BINARY_DIR}/zephyr-qemu-locore.elf
+    2>&1 | grep -iv \"empty loadable segment detected\" || true
```

Comment:
pipe and grep won't run on Windows.

Does qemu run on windows? If so we would want to do something else here
to avoid losing Windows support.

Reply:
> Does qemu run on windows?

As far as I can see, `qemu.cmake` is filled with all sorts of things that are not supported on Windows (e.g. "named pipes" under `/tmp`, `mkfifo`), so I guess not.

Moreover, with the WSL (and especially the upcoming WSL 2 that provides almost native performance), it is very questionable whether the native Windows environment support is really necessary for (arguably) advanced features like this.

Regarding `2>&1 | grep -iv \"empty loadable segment detected\" || true`, I do not see why this would not run on Windows.

`grep` can be easily installed using the Chocolatey (which is the recommended package manager for Windows in the Zephyr docs). `2>&1`, `|` and `||` are all valid and have the same meaning in the Windows shell. The only problem would be `true` if it ever executes, which should not normally be the case; I do not know if Chocolatey has a package that provides `true` and `false`, but one could easily download, or even make them in a few minutes (but then, WSL is just a few clicks away if you use Windows 10 ...).

Reply:
At first glance, it seems like there isn't even a toolchain available for building x86-64 Zephyr on the Windows (unless you are willing to compile `x86_64-elf` binutils and gcc yourself).

I think we can safely assume that nobody is running the x86-64 Zephyr on the QEMU on the Windows.

Reply:
Then this is fine, thank you for checking.